OCPBUGS-25761: Clarified relationship between rules and profiles (#89115)
openshift/openshift-docs: modules/compliance-profile-types.adoc

// Module included in the following assemblies:
//
// * security/compliance_operator/co-concepts/compliance-operator-understanding.adoc
// * security/compliance_operator/co-scans/compliance-operator-supported-profiles.adoc

:_mod-docs-content-type: REFERENCE
[id="compliance_profile_types_{context}"]
= Compliance Operator profile types

> Compliance Operator rules are organized into profiles. Profiles can target the {product-title} Platform or Nodes, and some standards include `rhcos4` Node profiles.

Reviewer: Consider rewriting this to avoid the phrase "OpenShift Container Platform Platform".
```suggestion
Compliance Operator rules are organized into profiles. Profiles can target the Platform or Nodes for {product-title}, and some standards include `rhcos4` Node profiles.

```

> Platform:: Platform profiles evaluate your {product-title} cluster components. For example, confirming APIServer configurations are using strong encryption cyphers is Platform-level rule.

Reviewer: ```suggestion
Platform:: Platform profiles evaluate your {product-title} cluster components. For example, a Platform-level rule can confirm whether APIServer configurations are using strong encryption cyphers.
```

> Node:: Node profiles for evaluating the OpenShift or {op-system} configuration of each host. There are `ocp4` Node profiles and `rhcos4` Node profiles. `ocp4` Node profiles evaluate the OpenShift configuration of each host. For example, confirming kubeconfig files have the correct permissions to meet a compliance standard is a Node-level rule. `rhcos4` Node profiles evaluate the {op-system-first} configuration of each host. For example, confirming the SSHD service is configured to disable password logins is a Node-level rule.

Reviewer: ```suggestion
Node:: Node profiles evaluate the OpenShift or {op-system} configuration of each host. You can use two Node profiles: `ocp4` Node profiles and `rhcos4` Node profiles. The `ocp4` Node profiles evaluate the OpenShift configuration of each host. For example, they can confirm whether `kubeconfig` files have the correct permissions to meet a compliance standard. The `rhcos4` Node profiles evaluate the {op-system-first} configuration of each host. For example, they can confirm whether the SSHD service is configured to disable password logins.

```

> Node:: Node profiles for evaluating the OpenShift or {op-system} configuration of each host. There are `ocp4` Node profiles and `rhcos4` Node profiles. `ocp4` Node profiles evaluate the OpenShift configuration of each host. For example, confirming kubeconfig files have the correct permissions to meet a compliance standard is a Node-level rule. `rhcos4` Node profiles evaluate the {op-system-first} configuration of each host. For example, confirming the SSHD service is configured to disable password logins is an {op-system}-level rule.

Reviewer 2: At the very end of this sentence we're saying `RHCOS-level`, while referring to `Node-level` earlier in the sentence.

Do we want to replace `{op-system}-level` with `Node-level` so it's consistent? If we stick with `{op-system}-level`, it's technically a new term we haven't defined.

[IMPORTANT]
====

> For benchmarks that have Node and Platform profiles, such as PCI-DSS, you must run both profiles in your {product-title} environment.

Reviewer 3: Is it better to use below sentence?
`For benchmarks that have `ocp4` Platform and `ocp4` Node profiles, such as PCI-DSS, you must run both profiles in your {product-title} environment.`

Author: I specifically worded it this way to make it clear to readers that regardless of whether a benchmark includes rhcos4 or ocp4 nodes, the profiles must be used with platform profiles. And I used PCI-DSS as an example because it was the first one I could think of that used both. 

I'd like to keep it this way, but we can make changes later if you think it's worth re-addressing later. 

In the meantime I'll go ahead and get this content peer reviewed and merged.

> For compliance standards that have Node and Platform profiles, such as PCI-DSS, you must run both profiles in your {product-title} environment.

Reviewer 2: Super minor nit: but `compliance standard` might be less familiar to readers than `benchmark` or `baseline`.

Similar comment on line 20.

For benchmarks that have `ocp4` Platform, `ocp4` Node, and `rhcos4` Node profiles, such as FedRAMP High, you must run all three profiles in your {product-title} environment.
====

> 

Reviewer: From the [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#admonitions): "Keep admonitions to a minimum, and avoid placing multiple admonitions close to one another. If multiple admonitions are necessary, restructure the information by moving the less-important statements into the flow of the main content."

[NOTE]
====

> Both `ocp4` Node and `rhcos4` Node scans may take a long time to complete in a cluster with many Nodes.

Reviewer: ```suggestion
In a cluster with many Nodes, both `ocp4` Node and `rhcos4` Node scans might take a long time to complete.
```
ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#may

> Both Node and `rhcos4` Node scans may take a long time to complete in a cluster with many Nodes.

Reviewer 2: nit: Similar comment to above. I'd clarify "Node" with `ocp4` Node so it has a more clear label about what that profile is checking.

> // Module included in the following assemblies:

Author: Resolved.

//
// * security/compliance_operator/co-concepts/compliance-operator-understanding.adoc
// * security/compliance_operator/co-scans/compliance-operator-supported-profiles.adoc

[id="compliance_profile_types_{context}"]
== Compliance Operator profile types

> There are three types of compliance profiles available: Platform Node, and {op-system-first} profiles. 

Reviewer 3: Better to be `There are three types of compliance profiles available: Platform, Node, and {op-system-first} profiles. `

Author: Good catch, thank you!

// Module included in the following assemblies:
//
// * security/compliance_operator/co-concepts/compliance-operator-understanding.adoc
// * security/compliance_operator/co-scans/compliance-operator-supported-profiles.adoc

:_mod-docs-content-type: REFERENCE
[id="compliance_profile_types_{context}"]
= Compliance Operator profile types

> There are three types of compliance profiles available: Platform, Node, and {op-system-first} profiles. These profile types contain rules that check both OpenShift and operating system settings, such as {op-system} configurations. 

Reviewer 2: This is a tricky thing to describe.

Technically - there are only two types of profiles. `Platform` and `Node`. But the ways in which we organize rules into profiles results in three different situations where you'd use a profile.

1.) Using a `Platform` profile to evaluate the OpenShift components, thus the platform
2.) Using a `Node` profile with OpenShift-specific rules to evaluate how OpenShift is configured across the entire cluster
3.) Using a `Node` profile with RHEL/RHCOS-specific rules to harden the underlying operating system running the OpenShift platform.

I'm not sure `{op-system}` is a clear replacement for the 3rd case though 🤔 .

Reviewer 3: The three types makes me confusion. I like the suggestion here.

Platform:: Platform profiles evaluate your {product-title} cluster components. For example, confirming APIServer configurations are using strong encryption cyphers is Platform-level rule.

> Node:: Node profiles for evaluating the OpenShift configuration of each host. For example, confirming kubeconfig files have the correct permissions to meet a compliance standard is a Node-level rule.

Reviewer 3: Still a little weird here.
How about below paragragh? 
```
Node: Node profiles for evaluating the OpenShift/{op-system} configuration of each host. 
There are ocp4 node profiles and rhcos4 node profiles. ocp4 node profiles mean to evaluating the OpenShift configuration of each host. For example, confirming kubeconfig files have the correct permissions to meet a compliance standard is a Node-level rule. rhcos4 node profiles are for  evaluating the {op-system-first} configuration of each host. For example, confirming the SSHD service is configured to disable password logins is an {op-system}-level rule.

```

> Node:: Node scans target the nodes of the cluster, checking every Node in a cluster to verify settings. For example, confirming kubeconfig files have the correct permissions to meet a compliance standard is a Node-level rule.

Reviewer 2: Since `Node` can mean different things depending on if it's for OpenShift or RHCOS, we could break this into two distinct pieces?

1. Platform profiles for evaluating OpenShift, irrespective of host (e.g., `APIServer`, `Scheduler`, etc.)
2. Node profiles for evaluating OpenShift configuration on each host
3. Node profiles for evaluating RHCOS configuration on each host
